modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_zephyr.c

```diff
@@ -201,3 +256,98 @@ uint64_t nrf_802154_platform_sl_lptimer_sync_lpticks_get(void)
 {
 	return m_sync_timer.target_time;
 }
+
+nrf_802154_sl_lptimer_platform_result_t nrf_802154_platform_sl_lptimer_hw_task_prepare(
+	uint64_t fire_lpticks,
+	uint32_t ppi_channel)
+{
+	uint32_t evt_address;
+	nrf_802154_sl_mcu_critical_state_t mcu_cs_state;
+
+	if (!hw_task_state_set(HW_TASK_STATE_IDLE, HW_TASK_STATE_SETTING_UP)) {
+		/* The only one available set of peripherals is already used. */
+		return NRF_802154_SL_LPTIMER_PLATFORM_NO_RESOURCES;
+	}
+
+	/* Allocate the channel if not already done. */
+	if (m_hw_task.chan == RTC_CHAN_INVALID) {
+		/* The channel allocation cannot take place during module
+		 * initialization, because for nRF53 it would run
+		 * out of resources - some other module temporarily needs
+		 * an rtc channel to initialize. For this reason, this is where
+		 * the allocation is made.
+		 * Once a channel has been successfully allocated, it will be held
+		 * until nrf_802154_platform_sl_lp_timer_deinit is called.
+		 */
+		m_hw_task.chan = z_nrf_rtc_timer_chan_alloc();
+		if (m_hw_task.chan < 0) {
+			m_hw_task.chan = RTC_CHAN_INVALID;
```

Comment:
This line could be removed, if instead of checking `if (m_hw_task.chan == RTC_CHAN_INVALID)` we would check `if (m_hw_task.chan < 0)`

These would save just a couple of bytes of flash, but as we have always some problems with flash, maybe these couple of bytes are worth it.

Reply:
I have the optimizations you suggested on this branch [ftr/ieee802154_hwtask_platf_optim](https://github.com/AdamZelikNS/sdk-nrf/tree/ftr/ieee802154_hwtask_platf_optim)
Adding them here would require time-consuming retesting for stability. Can we make an agreement that I will introduce these corrections (and adress all your comments from Jan 26) in the next PR in order to merge this one as soon as possible?